google/cloud/bigtable/testing/random_names.cc

```diff
@@ -0,0 +1,113 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/bigtable/testing/random_names.h"
+#include "google/cloud/internal/format_time_point.h"
+
+namespace google {
+namespace cloud {
+namespace bigtable {
+namespace testing {
+
+// Unless otherwise noted, the maximum ID lengths discovered by trial and error.
+auto constexpr kMaxTableIdLength = 50;
+char const kRandomTableIdRE[] = R"re(^tbl-\d{4}-\d{2}-\d{2}-.*$)re";
+static_assert(sizeof(kRandomTableIdRE) < kMaxTableIdLength,
```

Comment:
Comparing the length of a regexp to the length of something it might match seems like apples and oranges.

Reply:
Doh! You are right... I was not thinking.  Let me fix that.